demo/monolayer_growth/src/behaviours.h

```diff
@@ -0,0 +1,115 @@
+// -----------------------------------------------------------------------------
+//
+// Copyright (C) 2021 CERN & University of Surrey for the benefit of the
+// BioDynaMo collaboration. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+// See the LICENSE file distributed with this work for details.
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership.
+//
+// -----------------------------------------------------------------------------
+
+#ifndef BEHAVIOURS_H_
+#define BEHAVIOURS_H_
+
+#include "biodynamo.h"
+#include "core/behavior/behavior.h"
+#include "cycling_cell.h"
+#include "sim_param.h"
+
+namespace bdm {
+
+// Define growth behaviour
+struct Growth : public Behavior {
+  BDM_BEHAVIOR_HEADER(Growth, Behavior, 1);
+
+  Growth() { AlwaysCopyToNew(); }
+  virtual ~Growth() {}
+
+  void Run(Agent* agent) override {
+    if (auto* cell = dynamic_cast<CyclingCell*>(agent)) {
+      auto* random = Simulation::GetActive()->GetRandom();
+      real_t ran = random->Uniform(0, 1) * 1.0;
+      auto* sparam =
+          Simulation::GetActive()
+              ->GetParam()
+              ->Get<SimParam>();  // get a pointer to an instance of SimParam
+
+      // Counter for Delta t at each stage
+      // Used for calculating probability of moving to next state.
+      cell->SetDelt(cell->GetDelt() + 0.1);
+
+      // If statements for checking what states we are in and if
+      // a cell moves to the next state based on cumulative probability.
+      if (cell->GetCycle() == 1) {
+        real_t p1 = (cell->GetDelt() / 7) * cell->GetDelt();
+        if (p1 > ran) {
+          // Changing cells state number or "cycle" postiion.
+          cell->SetCycle(2);
+          // Delta t is always reset when exiting a state for use in the next
+          // state.
+          cell->SetDelt(0);
+        }
+
+      } else if (cell->GetCycle() == 2) {
+        real_t p2 = (cell->GetDelt() / 6) * cell->GetDelt();
+        if (p2 > ran) {
+          cell->SetCycle(3);
+          cell->SetDelt(0);
+        }
+
+      } else if (cell->GetCycle() == 3) {
+        real_t p3 = (cell->GetDelt() / 3) * cell->GetDelt();
```

Comment:
Can we add a comment for the cell cycle here? Actually for all lines of type 
```
real_t p3 = (cell->GetDelt() / N) * cell->GetDelt();
```
because I don't know why we divide here or what `GetDelt()^2/N` is. Maybe @BJackal can help here.

Reply:
Yes, we will need @BJackal help here

Reply:
Sorry for the later reply, didnt get a github ping. Ill look across everything.

Reply:
Here what we can do is replace the 3 with and N where N is the maximum time a cell spends in that state before switching to the next state. So for example. In total we have 4 phases: 0,1,2 and 3. I can add a class to the cell which is declared on agent creation. This then gives what the maximum time should be before a cell will enter the state.

Ill write this into the document so it can be see and add some comments.

Reply:
I thought the easiest way to do this would be create a member of the agent specific for each states maximum time in the form of an array. Some thing along the lines of double "f[4] = {7,6,3,2}". I suppose in this case it could be an int as all members are int values. What do you think ?

Reply:
Alright understood. I replaced the numbers with variables but we can hard code them. Thanks for clarification!